src/Sarif.Converters/TSLintConverter.cs

```diff
@@ -0,0 +1,144 @@
+            }
+
+            var fileInfoFactory = new FileInfoFactory(MimeType.DetermineFromFileExtension, loggingOptions);
+            Dictionary<string, FileData> fileDictionary = fileInfoFactory.Create(results);
+            
+            if (fileDictionary != null && fileDictionary.Any())
+            {
+                output.WriteFiles(fileDictionary);
+            }
+
+            output.OpenResults();
+            output.WriteResults(results);
+            output.CloseResults();
+        }
+
+        internal Result CreateResult(TSLintLogEntry tslEntry)
+        {
+            tslEntry = tslEntry ?? throw new ArgumentNullException(nameof(tslEntry));
+
+            Result result = new Result()
+            {
+                RuleId = tslEntry.RuleName,
+                Message = tslEntry.Failure
+            };
+
+            switch (tslEntry.RuleSeverity)
+            {
+                case "ERROR":
+                    result.Level = ResultLevel.Error;
+                    break;
+                case "WARN":
+                case "WARNING":
+                    result.Level = ResultLevel.Warning;
+                    break;
+                case "DEFAULT":
+                    result.Level = ResultLevel.Default;
+                    break;
+                default:
+                    result.Level = ResultLevel.NotApplicable;
+                    break;
+            }
+
+            result.Locations = new List<Location>();
+
+            Region region = new Region()
+            {
+                // The TSLint logs have line and column start at 0, Sarif has them starting at 1, so add 1 to each
+                StartLine = tslEntry.StartPosition.Line + 1,
+                StartColumn = tslEntry.StartPosition.Character + 1,
+                EndLine = tslEntry.EndPosition.Line + 1,
+                EndColumn = tslEntry.EndPosition.Character + 1,
+
+                Offset = tslEntry.StartPosition.Position
+            };
+
+            int length = tslEntry.EndPosition.Position - tslEntry.StartPosition.Position + 1;
+            region.Length = length > 0 ? length : 0;
+
+            Uri analysisTargetUri = new Uri(tslEntry.Name, UriKind.Relative);
+
+            PhysicalLocation analysisTarget = new PhysicalLocation(uri: analysisTargetUri, uriBaseId: null, region: region);
+            Location location = new Location();
+            location.AnalysisTarget = analysisTarget;
+
+            result.Locations.Add(location);
+
+            result.Fixes = new List<Fix>();
+
+            if (tslEntry.Fixes?.Any() == true)
+            {
+                IList<Replacement> replacements = new List<Replacement>();
+
+                foreach (TSLintLogFix fix in tslEntry.Fixes)
+                {
+                    Replacement replacement = new Replacement()
+                    {
+                        Offset = fix.InnerStart,
+                        DeletedLength = fix.InnerLength,
+                        InsertedBytes = fix.InnerText
```

Comment:
I didn't get a clarification on this part in the initial CR. Is InsertedBytes supposed to be pre-base64encoded, or does the writer take care of that? I can easily add
```var plainTextBytes = System.Text.Encoding.UTF8.GetBytes(fix.InnerText);```
```replacement.InsertedBytes = System.Convert.ToBase64String(plainTextBytes);```

Reply:
Hi Bogdan! The SARIF spec [says](https://rawgit.com/sarif-standard/sarif-spec/master/Static%20Analysis%20Results%20Interchange%20Format%20(SARIF).html#replacement-insertedbytes) that the inserted bytes must be Base64-encoded:

> If the file into which the bytes are to be inserted is a binary file, the value of the `insertedBytes` string shall be the MIME Base64 encoding of the byte sequence to be inserted.
>
> If the file into which the bytes are to be inserted is a text file, the characters to be inserted shall first be encoded in UTF-8. The value of the `insertedBytes` string shall be the MIME Base64 encoding of the resulting UTF-8 byte sequence.

Reply:
I see this is fixed.